gluten-core/src/main/scala/io/glutenproject/extension/GlutenPlan.scala

```diff
@@ -17,13 +17,77 @@
 
 package io.glutenproject.extension
 
+import scala.collection.JavaConverters._
+
+import io.glutenproject.GlutenConfig
+import io.glutenproject.expression.TransformerState
+import io.glutenproject.test.TestStats
+import io.glutenproject.utils.LogLevelUtil
+import io.glutenproject.validate.NativePlanValidatorInfo
+import org.apache.spark.sql.execution.SparkPlan
+
+case class ValidationResult(
+    validated: Boolean,
+    reason: Option[String])
+
+object ValidationResult {
+  def ok: ValidationResult = ValidationResult(validated = true, None)
+  def notOk(reason: String): ValidationResult = ValidationResult(validated = false, Some(reason))
```

Comment:
`reason` is not always a `String` here. For the validate log from native side, it's a `Vector`.

Reply:
yea, it has been flattened by `.mkString("native check failure:", ", ", "")`, see `Gluten.nativeValidationResult`